src/software/jetson_nano/services/network/network.cpp

```diff
@@ -22,9 +24,14 @@ std::tuple<TbotsProto::PrimitiveSet, TbotsProto::World> NetworkService::poll(
     const TbotsProto::RobotStatus& robot_status)
 {
     std::scoped_lock lock{primitive_set_mutex, world_mutex};
-    TbotsProto::RobotStatus new_status = robot_status;
-    new_status.set_last_handled_primitive_set(primitive_set_msg.sequence_number());
-    sender->sendProto(robot_status);
+    // Rate limit sending of proto based on thunderloop freq
+    if (network_ticks / (thunderloop_ticks + 1.0) <=
+        ROBOT_STATUS_BROADCAST_RATE_HZ / (CONTROL_LOOP_HZ + 1.0))
+    {
+        sender->sendProto(robot_status);
+        network_ticks = (network_ticks + 1) % ROBOT_STATUS_BROADCAST_RATE_HZ;
```

Comment:
This currently means that we lose some robot status messages. Do we want to merge the key components of the not sent messages with the one that is going to be sent? E.g. Not sure if this is possible, but if we have an `ErrorCode` in one message which doesnt get sent, and then the `ErrorCode` is erased in the next message which does get sent. This could possibly be problematic...

Reply:
i think most errors persist? at least for motor status wise

Reply:
ya Im not sure. Since I think we reset the motor controller depending on the error as well. But I like your current change of sending the msg if theres a fault